master/internal/api_search.go

```diff
@@ -214,3 +214,230 @@ func (a *apiServer) PutSearchRetainLogs(
+}
+
+func (a *apiServer) KillSearch(ctx context.Context, req *apiv2.KillSearchRequest,
+) (*apiv2.KillSearchResponse, error) {
+	expReq := apiv1.KillExperimentRequest{
+		Id: req.Id,
+	}
+	_, err := a.KillExperiment(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.KillSearchResponse{}
+	return &res, nil
+}
+
+func (a *apiServer) KillSearches(ctx context.Context, req *apiv2.KillSearchesRequest,
+) (*apiv2.KillSearchesResponse, error) {
+	expReq := apiv1.KillExperimentsRequest{
+		ExperimentIds: req.SearchIds,
+		Filters:       bulkFiltersSearchToExperiment(req.Filters),
+		ProjectId:     req.ProjectId,
+	}
+	expRes, err := a.KillExperiments(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.KillSearchesResponse{
+		Results: expRes.Results,
+	}
+	return &res, nil
+}
+
+func (a *apiServer) ArchiveSearch(ctx context.Context, req *apiv2.ArchiveSearchRequest,
+) (*apiv2.ArchiveSearchResponse, error) {
+	expReq := apiv1.ArchiveExperimentRequest{
+		Id: req.Id,
+	}
+	_, err := a.ArchiveExperiment(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.ArchiveSearchResponse{}
+	return &res, nil
+}
+
+func (a *apiServer) ArchiveSearches(ctx context.Context, req *apiv2.ArchiveSearchesRequest,
+) (*apiv2.ArchiveSearchesResponse, error) {
+	expReq := apiv1.ArchiveExperimentsRequest{
+		ExperimentIds: req.SearchIds,
+		Filters:       bulkFiltersSearchToExperiment(req.Filters),
+		ProjectId:     req.ProjectId,
+	}
+	expRes, err := a.ArchiveExperiments(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.ArchiveSearchesResponse{
+		Results: expRes.Results,
+	}
+	return &res, nil
+}
+
+func (a *apiServer) UnarchiveSearch(ctx context.Context, req *apiv2.UnarchiveSearchRequest,
+) (*apiv2.UnarchiveSearchResponse, error) {
+	expReq := apiv1.UnarchiveExperimentRequest{
+		Id: req.Id,
+	}
+	_, err := a.UnarchiveExperiment(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.UnarchiveSearchResponse{}
+	return &res, nil
+}
+
+func (a *apiServer) UnarchiveSearches(ctx context.Context, req *apiv2.UnarchiveSearchesRequest,
+) (*apiv2.UnarchiveSearchesResponse, error) {
+	expReq := apiv1.UnarchiveExperimentsRequest{
+		ExperimentIds: req.SearchIds,
+		Filters:       bulkFiltersSearchToExperiment(req.Filters),
+		ProjectId:     req.ProjectId,
+	}
+	expRes, err := a.UnarchiveExperiments(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.UnarchiveSearchesResponse{
+		Results: expRes.Results,
+	}
+	return &res, nil
+}
+
+func (a *apiServer) DeleteSearch(ctx context.Context, req *apiv2.DeleteSearchRequest,
+) (*apiv2.DeleteSearchResponse, error) {
+	expReq := apiv1.DeleteExperimentRequest{
+		ExperimentId: req.SearchId,
+	}
+	_, err := a.DeleteExperiment(ctx, &expReq)
+	if err != nil {
+		return nil, err
+	}
+	res := apiv2.DeleteSearchResponse{}
+	return &res, nil
+}
+
+func (a *apiServer) DeleteSearches(ctx context.Context, req *apiv2.DeleteSearchesRequest,
```

Comment:
tests?

Reply:
These endpoints are essentially wrappers around already tested endpoints. Added a test for the newly added bulk conversion filter function